app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala

```diff
@@ -23,19 +42,32 @@ class GovernmentGatewayProxyConnector(baseUrl: URL, httpPost: HttpPost) {
   }
 
   def parseResponse(xmlString: String) : Option[AgentDetails] = {
-    val XML = scala.xml.XML.withSAXParser(SAXParserFactory.newInstance().newSAXParser())
-
-    val xml = XML.loadString(xmlString)
+    val xml: Elem = toXmlElement(xmlString)
     val agentDetails = xml \ "AllocatedAgents" \ "AgentDetails"
-    val agentCode = (agentDetails \ "AgentCode").head.text
+    if (agentDetails.nonEmpty) {
+      val agentCode = (agentDetails \ "AgentCode").head.text
 
-    val credentials = (agentDetails \ "AssignedCredentials" \ "Credential").map { elem =>
-      AssignedCredentials((elem \ "CredentialIdentifier").head.text)
+      val credentials = (agentDetails \ "AssignedCredentials" \ "Credential").map { elem =>
+        AssignedCredentials((elem \ "CredentialIdentifier").head.text)
+      }
+      Some(AgentDetails(agentCode, credentials))
```

Comment:
It was a fairly conscious decision to not handle this case. We don't know what the shape of the data is, or if it's possible from a business POV. A question definitely to ask the GG team. I suggest we defer any changes until we know what the data look like.